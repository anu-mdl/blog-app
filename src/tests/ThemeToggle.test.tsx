import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ThemeToggle from '@/components/header/ThemeToggle';

jest.mock('next-themes', () => ({
  useTheme: jest.fn()
}));

jest.mock('lucide-react', () => ({
  Sun: () => <svg data-testid="sun-icon" />,
  Moon: () => <svg data-testid="moon-icon" />
}));

jest.mock('@/components/ui/button', () => ({
  Button: ({ children, ...props }: any) => (
    <button {...props}>{children}</button>
  )
}));

describe('ThemeToggle', () => {
  const mockSetTheme = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches light theme snapshot', () => {
    require('next-themes').useTheme.mockImplementation(() => ({
      theme: 'light',
      setTheme: mockSetTheme
    }));

    const { asFragment } = render(<ThemeToggle />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches dark theme snapshot', () => {
    require('next-themes').useTheme.mockImplementation(() => ({
      theme: 'dark',
      setTheme: mockSetTheme
    }));

    const { asFragment } = render(<ThemeToggle />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('toggles theme on click', async () => {
    require('next-themes').useTheme.mockImplementation(() => ({
      theme: 'light',
      setTheme: mockSetTheme
    }));

    render(<ThemeToggle />);
    await userEvent.click(screen.getByRole('button'));

    expect(mockSetTheme).toHaveBeenCalledWith('dark');
  });
});
