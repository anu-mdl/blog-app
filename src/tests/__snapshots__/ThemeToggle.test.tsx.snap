// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggle matches dark theme snapshot 1`] = `
<DocumentFragment>
  <button
    class="hidden sm:inline-flex"
    variant="ghost"
  >
    <svg
      data-testid="sun-icon"
    />
    <svg
      data-testid="moon-icon"
    />
    <span
      class="sr-only"
    >
      Toggle theme
    </span>
  </button>
</DocumentFragment>
`;

exports[`ThemeToggle matches light theme snapshot 1`] = `
<DocumentFragment>
  <button
    class="hidden sm:inline-flex"
    variant="ghost"
  >
    <svg
      data-testid="sun-icon"
    />
    <svg
      data-testid="moon-icon"
    />
    <span
      class="sr-only"
    >
      Toggle theme
    </span>
  </button>
</DocumentFragment>
`;
